integration/test/Testlib/Certs.hs

```diff
@@ -0,0 +1,146 @@
+module Testlib.Certs where
+
+import Crypto.Hash.Algorithms (SHA256 (SHA256))
+import qualified Crypto.PubKey.RSA as RSA
+import qualified Crypto.PubKey.RSA.PKCS15 as PKCS15
+import Crypto.Store.PKCS8 (PrivateKeyFormat (PKCS8Format), keyToPEM)
+import Crypto.Store.X509 (pubKeyToPEM)
+import Data.ASN1.OID (OIDable (getObjectID))
+import Data.Hourglass
+import Data.PEM (PEM (PEM), pemWriteBS)
+import Data.String.Conversions (cs)
+import Data.X509
+import Testlib.Prelude
+
+type RSAKeyPair = (RSA.PublicKey, RSA.PrivateKey)
+
+type SignedCert = SignedExact Certificate
+
+-- | convert a PEM to a string
+toPem :: PEM -> String
+toPem = cs . pemWriteBS
+
+-- | convert a signed certificate to a string
+signedCertToString :: SignedCert -> String
+signedCertToString = toPem . PEM "CERTIFICATE" [] . encodeSignedObject
+
+-- | convert a private key to string
+privateKeyToString :: RSA.PrivateKey -> String
+privateKeyToString = toPem . keyToPEM PKCS8Format . PrivKeyRSA
+
+-- | convert a public key to string
+publicKeyToString :: RSA.PublicKey -> String
+publicKeyToString = toPem . pubKeyToPEM . PubKeyRSA
+
+-- | order: publickey, private key
+keyPairToString :: RSAKeyPair -> (String, String)
+keyPairToString = bimap publicKeyToString privateKeyToString
+
+-- | the minimum key size is hard coded to be 256 bytes (= 2048 bits)
+--
+-- TODO(mangoiv): key generation takes an (actually multiple) eternities
```

Comment:
Has this TODO been resolved?

Reply:
Yes, by using constant p and q for RSA keys. Should be fine for tests.

Reply:
Let's delete the comment then :)